pkg/core/manager/insight/summary.go

```diff
@@ -38,9 +38,10 @@ const (
 func (i *InsightManager) GetDetailsForCluster(ctx context.Context, client *multicluster.MultiClusterClient, name string) (*ClusterDetail, error) {
 	// get server version and measure latency
 	start := time.Now()
+	serverVersionStr := "v0.0.0-unknown" // default version when error occurs
 	serverVersion, err := client.ClientSet.DiscoveryClient.ServerVersion()
```

Comment:
The error case is not properly handled. Consider initializing `serverVersionStr` inside the if block and setting the default value in the else case for clarity.